### PR TITLE
[#84] points of interest table

### PR DESCRIPTION
**Pull request by jfacorro**
#84

### CODE REVIEW

File: src/erlang_ls_parser.erl [hunk 1]
```diff
@@ -60,8 +60,8 @@ parse_extra(IoDevice, Extra, StartLocation) ->
       end;
     {eof, _} ->
       {ok, Extra};
-    {error, Error} ->
-      {error, Error}
+    {error, ErrorInfo, EndLocation} ->
```
Comment (jfacorro) [inline src/erlang_ls_parser.erl]: This change is based on the error tuple that is returned from the call to `io:scan_erl_form/3`.

File: src/erlang_ls_tree.erl [hunk 2]
```diff
@@ -101,7 +101,7 @@ points_of_interest(Tree, Extra) ->
   try points_of_interest(Tree, Type, Extra)
   catch
     Class:Reason:Stacktrace ->
-      lager:warning( "Could not analyze tree: ~p:~p ~P"
+      lager:warning( "Could not analyze tree: ~p:~p ~p"
```
Comment (jfacorro) [inline src/erlang_ls_tree.erl]: `~P` seems to be an invalid format argument.

File: test/prop_statem.erl [hunk 3]
```diff
@@ -97,7 +97,7 @@ did_open(Uri, LanguageId, Version, Text) ->
   erlang_ls_client:did_open(Uri, LanguageId, Version, Text).
 
 did_open_args(_S) ->
-  [erlang_ls_proper_gen:uri(), <<"erlang">>, 0, utf8()].
+  [erlang_ls_proper_gen:uri(), <<"erlang">>, 0, erlang_ls_proper_gen:tokens()].
```
Comment (jfacorro) [inline test/prop_statem.erl]: Otherwise the parsing of the file fails.

File: src/erlang_ls_protocol_impl.erl [hunk 4]
```diff
@@ -124,8 +124,9 @@ textdocument_didchange(Params) ->
   case ContentChanges of
     []                      -> ok;
     [#{<<"text">> := Text}] ->
-      {ok, Buffer} = erlang_ls_buffer_server:get_buffer(Uri),
-      ok = erlang_ls_buffer:set_text(Buffer, Text)
+      {ok, Buffer0} = erlang_ls_db:find(erlang_ls_files, Uri),
```
Comment (jfacorro) [inline src/erlang_ls_protocol_impl.erl]: These three calls could be simplified with `erlang_ls_buffer:update(Uri, TexT)` if `erlang_ls_buffer` was the one writing to the table and in general the one that know what table should it use.

File: src/erlang_ls_tree.erl (hunk 2 above)
Comment (robertoaloi) [reply src/erlang_ls_tree.erl]: The depth argument was missing.

**Pull request merged**